### PR TITLE
new feature to filter connected field's options based off another con…

### PR DESCRIPTION
**Pull request by jamesduncan**
…nected field's selection

### CODE REVIEW

**Comment by Achoobert** [comment]
```
fetch
if we are filtering based off another selectivity's value we
need to do it on fetch each time because the value can change
```

Will this take extra time to load? 
Or is the full datacollection already local?

**Comment by jamesduncan** [comment]
We already have to make a fetch for each load. There are possible filters added to the fetch already so I am just adding new filters before the fetch is made. If you want to test it Isaac there is a branch out there that I have Pong reviewing. 

James Duncan

File: assets/opstools/AppBuilder/classes/platform/dataFields/ABFieldConnect.js [hunk 1]
```diff
@@ -814,22 +814,67 @@ module.exports = class ABFieldConnect extends ABFieldConnectCore {
...
+                  // need to do it on fetch each time because the value can change
+                  var self = this;
+                  // copy the filters so we don't add to them every time there is a change
+                  var combineFilters = JSON.parse(
+                     JSON.stringify(options.filters)
+                  );
+                  // only add filters if we pass valid value and key
+                  if (options.filterValue && options.filterKey) {
+                     // get the current value of the parent select box
+                     let parentVal = self.getValue($$(options.filterValue));
```
Comment (wongpratan) [inline assets/opstools/AppBuilder/classes/platform/dataFields/ABFieldConnect.js:864]: Are we able to use `this` directly instead of the `self` variable ?

Comment (jamesduncan) [reply assets/opstools/AppBuilder/classes/platform/dataFields/ABFieldConnect.js:864]: I can give it a try...I don't know why I am setting `var self = this;` on line 856. I will test it out and push changes if it works.

**Review by wongpratan — APPROVED** [review]

**Pull request merged**